roles/acme_certificate/tasks/dns-openstack-cleanup.yml

```diff
@@ -0,0 +1,18 @@
+---
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2022, Lucca Jiménez Könings (https://github.com/jimkoen)
+
+# Clean up DNS challenges for DNS provider Openstack
+- name: Cleaning up challenge DNS entries for {{ ', '.join(acme_certificate_domains + acme_certificate_ips) }} via OpenStack
+  openstack.cloud.recordset:
+    state: absent
+    # Openstack expects a zone name with a trailing dot
+    zone: "{{ acme_certificate_openstack_zone | default(item.key | community.dns.get_registrable_domain) }}."
```

Comment:
What is `acme_certificate_openstack_zone`? I don't see it documented anywhere.

Also when using `item.key`, you need to use `felixfontein.acme._substitute_dns(acme_certificate_dns_substitution)` as well (see the other DNS providers in the `main` branch).